cmake/onnxruntime_providers_iree.cmake

```diff
@@ -28,23 +28,24 @@ target_link_libraries(onnxruntime_providers_iree PRIVATE
   onnx
   protobuf::libprotobuf
 )
+if ( CMAKE_COMPILER_IS_GNUCC )
```

Comment:
This variable appears to be deprecated in newer releases of CMake, so we might want to replace it with some newer alternative like `CMAKE_C_COMPILER_ID`.

Would you want this code to run only for GCC? Is there a reason for this?

Reply:
Yeah, the flags here aren't recognized on windows cl. I'll look for a modern alternative to this.